consensus/src/dag/dag_state_sync.rs

```diff
@@ -242,18 +242,28 @@ impl DagStateSynchronizer {
         sync_dag_store: Arc<DagStore>,
         commit_li: LedgerInfoWithSignatures,
     ) -> anyhow::Result<DagStore> {
-        match dag_fetcher
-            .fetch(request, responders, sync_dag_store.clone())
-            .await
-        {
-            Ok(_) => {},
-            Err(err) => {
-                error!("error fetching nodes {}", err);
-                bail!(err)
-            },
-        }
+        let dag_store = sync_dag_store.clone();
+        let dag_sync_fut = async move {
+            debug!("Syncing DAG. Fetching Nodes");
+            dag_fetcher
+                .fetch(request.clone(), responders, dag_store.clone())
+                .await
+                .map_err(|err| {
+                    error!("error fetching nodes {}", err);
+                    anyhow!(err)
+                })?;
 
-        self.execution_client.sync_to(commit_li).await?;
+            Ok(())
+        };
+        let execution_client = self.execution_client.clone();
+        let state_sync_fut = async move {
+            debug!(target_ledger_info = commit_li, "Requesting sync to");
+            execution_client
+                .sync_to(commit_li)
+                .await
+                .map_err(|err| anyhow!(err))
+        };
+        futures::future::try_join(dag_sync_fut, state_sync_fut).await?;
```

Comment:
I think we discuss this can work, but I forget about how the race condition resolves, probably should add some explanation here why it's safe when one of them fails

Reply:
added a todo. will fill in later